src/driver/DriverUtils.ts

```diff
@@ -134,12 +134,27 @@ export class DriverUtils {
         let port = undefined;
         let hostReplicaSet = undefined;
         let replicaSet = undefined;
-        // remove mongodb query params
+        
+        let optionsObject: any = {};
```

Comment:
If `optionsObject` is annotated as `Record<string, string>` instead of `any`, it can be safely merged into `connectionUrl` without widening the resulting type.

```suggestion
        let optionsObject: Record<string, string> = {};
```